shell/platform/tizen/channels/text_input_channel.cc

```diff
@@ -303,92 +307,62 @@ void TextInputChannel::SendStateUpdate(const TextInputModel& model) {
-  bool is_ime = true;
-  // FIXME: Only for wearable.
-  if (is_ime && strcmp(event->key, "Select") == 0) {
-    is_in_select_mode_ = true;
-    FT_LOG(Debug) << "Entering select mode.";
-  }
-#else
-  const char* device_name = ecore_device_name_get(event->dev);
-  bool is_ime = device_name ? strcmp(device_name, "ime") == 0 : true;
-#endif
-
-  if (ShouldNotFilterEvent(event->key, is_ime)) {
-    FT_LOG(Info) << "Force redirect an IME key event: " << event->keyname;
-    Reset();
-    return false;
-  }
-
-  handled =
-      input_method_context_->FilterEvent(event, is_ime ? "ime" : "", is_down);
-
-#ifdef WEARABLE_PROFILE
-  if (!handled && !strcmp(event->key, "Return") && is_in_select_mode_) {
-    is_in_select_mode_ = false;
-    handled = true;
-    FT_LOG(Debug) << "Leaving select mode.";
-  }
-#endif
-
-  return handled;
-}
-
-void TextInputChannel::HandleUnfilteredEvent(Ecore_Event_Key* event) {
-  bool select = !strcmp(event->key, "Select");
-  bool shift = event->modifiers & ECORE_SHIFT;
+void TextInputChannel::HandleUnfilteredEvent(const char* key,
+                                             const char* string,
+                                             uint32_t modifires) {
+  bool shift = modifires & ECORE_SHIFT;
   bool needs_update = false;
-  std::string key = event->key;
+  std::string key_str = key;
 
-  if (key == "Left") {
+  if (key_str == "Left") {
     if (shift) {
       TextRange selection = active_model_->selection();
       needs_update = active_model_->SetSelection(
           TextRange(selection.base(), selection.extent() - 1));
     } else {
       needs_update = active_model_->MoveCursorBack();
     }
-  } else if (key == "Right") {
+  } else if (key_str == "Right") {
     if (shift) {
       TextRange selection = active_model_->selection();
       needs_update = active_model_->SetSelection(
           TextRange(selection.base(), selection.extent() + 1));
     } else {
       needs_update = active_model_->MoveCursorForward();
     }
-  } else if (key == "End") {
+  } else if (key_str == "End") {
     if (shift) {
       needs_update = active_model_->SelectToEnd();
     } else {
       needs_update = active_model_->MoveCursorToEnd();
     }
-  } else if (key == "Home") {
+  } else if (key_str == "Home") {
     if (shift) {
       needs_update = active_model_->SelectToBeginning();
     } else {
       needs_update = active_model_->MoveCursorToBeginning();
     }
-  } else if (key == "BackSpace") {
+  } else if (key_str == "BackSpace") {
     needs_update = active_model_->Backspace();
-  } else if (key == "Delete") {
+  } else if (key_str == "Delete") {
     needs_update = active_model_->Delete();
-  } else if (event->string && strlen(event->string) == 1 &&
-             IsAsciiPrintableKey(event->string[0])) {
-    active_model_->AddCodePoint(event->string[0]);
+  } else if (string && strlen(string) == 1 && IsAsciiPrintableKey(string[0])) {
+    active_model_->AddCodePoint(string[0]);
```

Comment:
Text input in languages ​​other than English that requires composing by IME, such as Korean or Japanese, is already supported.

This part you mentioned is inevitably added logic because the behavior of 'IME' is different in the profile except for TV.
(IME does not filter the key events corresponding to the ASCII in Tizen except TV profile.)

I will contact you via messenger for additional explanations.